### PR TITLE
🛠 Fix toFormattedString for null nodes

### PR DESCRIPTION
**Pull request by AyhamAl-Ali**
### Description

This PR fixes logger level color for messages that doesn't have a node.

Before:
image

After:

image

---
**Target Minecraft Versions:**  Any
**Requirements:**  None
**Related Issues:**  None

### CODE REVIEW

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/log/LogEntry.java [hunk 1]
```diff
@@ -159,15 +157,20 @@ public String toFormattedString() {
...
-
 		// Replace configured messages chat styles without user variables
 		String lineInfoMsg = replaceNewline(Utils.replaceEnglishChatStyles(lineInfo.getValue() == null ? lineInfo.key : lineInfo.getValue()));
 		String detailsMsg = replaceNewline(Utils.replaceEnglishChatStyles(details.getValue() == null ? details.key : details.getValue()));
 		String lineDetailsMsg = replaceNewline(Utils.replaceEnglishChatStyles(LINE_DETAILS.getValue() == null ? LINE_DETAILS.key : LINE_DETAILS.getValue()));
 
+		if (node == null)
+			return String.format(detailsMsg.replaceAll("(\\s){2,}", ""), message); // Remove line beginning spaces
+
+		Config c = node.getConfig();
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/log/LogEntry.java:168]:
```suggestion
		Config config = node.getConfig();
```

File: src/main/java/ch/njol/skript/log/LogEntry.java [hunk 2]
```diff
@@ -159,15 +157,20 @@ public String toFormattedString() {
...
-		if (!from.isEmpty())
-			from = "§7   " + from + "\n";
-
 		// Replace configured messages chat styles without user variables
 		String lineInfoMsg = replaceNewline(Utils.replaceEnglishChatStyles(lineInfo.getValue() == null ? lineInfo.key : lineInfo.getValue()));
 		String detailsMsg = replaceNewline(Utils.replaceEnglishChatStyles(details.getValue() == null ? details.key : details.getValue()));
 		String lineDetailsMsg = replaceNewline(Utils.replaceEnglishChatStyles(LINE_DETAILS.getValue() == null ? LINE_DETAILS.key : LINE_DETAILS.getValue()));
 
+		if (node == null)
+			return String.format(detailsMsg.replaceAll("(\\s){2,}", ""), message); // Remove line beginning spaces
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/log/LogEntry.java:166]: Not sure if replacing 2+ whitespace is the best approach here.

Also the group isn't needed I think: `\\s{2,}`

File: src/main/java/ch/njol/skript/log/LogEntry.java [hunk 3]
```diff
@@ -159,15 +157,20 @@ public String toFormattedString() {
...
+
+		Config c = node.getConfig();
+		String from = this.from;
+
+		if (!from.isEmpty())
+			from = "§7   " + from + "\n";
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/log/LogEntry.java:172]: I think we may prefer to use a ChatColor constant here

File: src/main/java/ch/njol/skript/log/LogEntry.java (hunk 2 above)
Comment (UnderscoreTud) [reply src/main/java/ch/njol/skript/log/LogEntry.java:165]: Adding upon what TP said, you can use `String#stripLeading` to remove the leading whitespaces in a string

**Review by UnderscoreTud — APPROVED** [review]

File: src/main/java/ch/njol/skript/log/LogEntry.java (hunk 2 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/log/LogEntry.java:165]: `String#stripLeading` Java 12 feature :(

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**